### PR TITLE
hotfix/template-name-fail-fix

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -718,7 +719,7 @@ Resources:
         - Key: IncludedInBackup
           Value: 'true'
 
-  NvaPersonInfoTable:
+  NvaIdTypeObjectsTable:
```
Comment (torbjokv) [inline template.yaml:722]: heads up, this can delete table, but I guess its ok

File: template.yaml [hunk 2]
```diff
@@ -718,7 +719,7 @@ Resources:
...
     Type: AWS::DynamoDB::Table
     UpdateReplacePolicy: Retain
     DeletionPolicy: Delete
```
Comment (torbjokv) [inline template.yaml:725]: what does nva-id-type-persistent-objects mean btw?

File: template.yaml (hunk 1 above)
Comment (torbjokv) [reply template.yaml:722]: or leave garbage, then we need to manually tidy in every env

File: user-access-service/src/test/java/no/unit/nva/database/TermsAndConditionsServiceTest.java [hunk 3]
```diff
@@ -20,6 +20,9 @@ public class TermsAndConditionsServiceTest {
     static void initialize() {
         var client = DatabaseTestConfig
                 .getEmbeddedClient();
+        var TABLE_NAME = new Environment()
```
Comment (torbjokv) [inline user-access-service/src/test/java/no/unit/nva/database/TermsAndConditionsServiceTest.java:23]: this is a test, do you need to use environment here at all?

File: user-access-service/src/test/java/no/unit/nva/database/DynamoCrudServiceTest.java [hunk 4]
```diff
@@ -15,18 +16,20 @@
...
-    private static DynamoCrudService<TermsConditions> termsConditionsService;
+    private static DynamoCrudService<TermsConditions> dynamoCrudService;
 
 
     @BeforeAll
     static void initialize() {
+
         var client = DatabaseTestConfig
                 .getEmbeddedClient();
+        var tableName = new Environment()
```
Comment (torbjokv) [inline user-access-service/src/test/java/no/unit/nva/database/DynamoCrudServiceTest.java:27]: I would just hard code this name like before?

File: template.yaml (hunk 1 above)
Comment (StigNorland) [reply template.yaml:722]: Table is new and unused, but thanks for warning me!

File: template.yaml (hunk 2 above)
Comment (StigNorland) [reply template.yaml:725]: it is a reference to the single-table-pattern used.
`id` and `type` as a composite key access pattern.

Comment (torbjokv) [reply template.yaml:725]: Hmmm I would prefer if it have a name that reflects its content/group of content, this is in my head way to generic. if we already are changing logic id here, its a good time to change table name also.

Comment (StigNorland) [reply template.yaml:725]: so evey entity can have a host of classes assosiated with it, but only one instanse of each class.

| id   | type            | createdBy | created     | modifiedBy | modified    | ...objectfields |
|------|-----------------|-----------|-------------|------------|-------------|-----------------|
| xyz  | Person          | xyz       | 10-01-2021  | xyz        | 01-01-2022  | personinfo....  |
| xyz  | Roles           | admin     | 00-01-2021  | admin      | 01-01-2023  | RolesInfo....   |
| xyz  | UserPreferences | admin     | 00-01-2021  | xyz        | 01-10-2024  | RolesInfo....   |

Comment (torbjokv) [reply template.yaml:724]: probably do a specific table name in a later pr

File: user-access-service/src/main/java/no/unit/nva/database/TermsAndConditionsService.java [hunk 5]
```diff
@@ -33,18 +33,22 @@ public TermsAndConditionsService(DynamoDbClient client, String tableName) {
...
+                                      .fetch(crudService)
+                                      .termsConditionsUri();
+                       } catch (NotFoundException e) {
+                           return null;
+                       }
+                   })
+                   .map(fetchedUri -> TermsConditionsResponse.builder()
+                                          .withTermsConditionsUri(fetchedUri)
+                                          .build())
+                   .orElse(null);
```
Comment (StigNorland) [inline user-access-service/src/main/java/no/unit/nva/database/TermsAndConditionsService.java:51]: 🥇

File: user-access-service/src/main/java/no/unit/nva/database/TermsAndConditionsService.java [hunk 6]
```diff
@@ -33,18 +33,22 @@ public TermsAndConditionsService(DynamoDbClient client, String tableName) {
     }
 
     public TermsConditionsResponse getTermsAndConditionsByPerson(URI cristinId) {
```
Comment (StigNorland) [inline user-access-service/src/main/java/no/unit/nva/database/TermsAndConditionsService.java:35]: cristinPersonId

**Review by torbjokv — APPROVED** [review]

**Pull request merged**